node/src/components/linear_chain_sync.rs

```diff
@@ -86,7 +86,7 @@ enum State {
         /// Linear chain block being downloaded.
         linear_chain_block: Box<Option<BlockHeader>>,
         /// Block we received from a node and are currently executing.
-        current_block: Box<Option<BlockHeader>>,
+        current_block: Box<BlockHeader>,
```

Comment:
If I understand correctly, it's not a _current block_ anymore but the last block we had successfully downloaded and handled.

Reply:
I didn't change when it gets set, except that I also set it when transitioning to `SyncingDescendants`.
Actually, the comment on the other `current_block` above also isn't correct, is it? `current_block` is not set when we start downloading it, but when we download its _deploys_. And it stays until we download the next block's deploys.

I still think the name is kind of okay; not sure what's the alternative? `latest_block`?

Also, do we need `linear_chain_block` at all?